sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NamedRelation.scala

```diff
@@ -21,4 +21,7 @@ import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 
 trait NamedRelation extends LogicalPlan {
   def name: String
+
+  // When true, the schema of input data must match the schema of this relation, during write.
+  def requireSchemaMatch: Boolean = true
```

Comment:
Can we use `def ignoreSchemaMatch: Boolean = false` because we have only one instance and it's negative form.
```scala
- !table.requireSchemaMatch || {
+ table.ignoreSchemaMatch || {
```

Reply:
`ignoreSchemaMatch` sounds a little weird. I don't have a strong opinion about negative form vs positive form, but the naming must be precise here.

Reply:
I agree that the "ignore" form is better.

I don't think that the term "schema match" is clear though, because it is not obvious what "match" means. What about `skipSchemaResolution` or `skipSchemaValidation`?